src/GridView/HierarchyStepper.js

```diff
@@ -3,19 +3,72 @@ import { Step, Stepper, StepLabel } from 'material-ui/Stepper';
 
 
 class HierarchyStepper extends Component {
+	constructor() {
+		super();
+		this.entityTypes = [];
+		this.state = {
+			highlitedEntityTypes: []
+		}
+	}
+
+	componentDidMount() {
+		this.updateEntityTypes();
+		this.setState({
+			highlitedEntityTypes: this.getHighlitedEntityTypes(this.props.currentEntityTypeId)
+		});
+
+	}
+
+	updateEntityTypes() {
+		this.props.hierarchy.forEach((hierarchyLayer) => {
+			hierarchyLayer.forEach((entityType) => {
```

Comment:
We can avoid the second for-loop by concatenating  entityType-Arrays of hierarchyLayers.